public/js/lib/colorContraster.js

```diff
@@ -31,12 +31,44 @@ function parseColor(color) {
 
 function parseHex(color) {
   // #ff00de
-  // support for 3 character hex strings would be nice :)
-  // pull request me if you like - noopkat
-  const justColor = color.substring(1, 7);
-  const r = parseInt(justColor.substring(0, 2), 16); // hexToR
-  const g = parseInt(justColor.substring(2, 4), 16); // hexToG
-  const b = parseInt(justColor.substring(4, 6), 16); // hexToB
+  
+  // adding a simple error check in case of an invalid color string
+  // check if string is either 4 or 7 characters long and is string
+  // added 9 for #RGBA types value (we remove alpha part later using substring) 
+  // Also, RGBA don't have short code
+  if (!([4, 7, 9].includes(color.length)) || typeof color !== "string") {
+    return null;
+  }
+  // Another small check to validate valid HEX input
+  // 1. Remove '#' from string
+  // 2. Split into array of chars
+  // 3. Map over each entry to check if parseInt with base 16 yields valid number
+  // 4. Finally test if a NaN is present.
+  // If NaN were present, this condition will be true implying invalid Hex code
+  // It might be an overkill here
+  if (Array.from(color.substr(1))
+       .map(b => isNaN(parseInt(b, 16)))
```

Comment:
This works, but could be done in 1 step with `.find(char => isNaN(parseInt(char, 16)))`. Given how large this function is now, I'd extract the validation to an `validHexColor()` function.

Reply:
Sure Let me update that